core/ir.ml

```diff
@@ -75,7 +75,13 @@ and binding =
   | `Module of (string * binding list option) ]
 and special =
   [ `Wrong of Types.datatype
-  | `Database of value
+  | `Database of value 
+  | `Lens of value * Types.lens_sort
+  | `LensDrop of value * string * string * value * Types.lens_sort
+  | `LensSelect of value * Sugartypes.phrase * Types.lens_sort
+  | `LensJoin of value * value * string list * Sugartypes.phrase * Sugartypes.phrase * Types.lens_sort
```

Comment:
This doesn't look right. Why haven't `Sugartypes.phrase`s been translated? The IR should not contain instances of concrete syntax.

Reply:
Hm yes one should be able to use `lens_phrase` at this point. I'll fix this tomorrow.

Reply:
Fixed this.